### DISCUSSION TITLE
How to detect when Hydratation is done in javascript ?

### DISCUSSION THREAD

**Discussion by GuyoST** [remix-run/remix#6343] [Q&A] [answered]
Hello every one,

I have a strange problem for a website (I'm using remix now for this website, it was an existing website).
This website uses some jQuery plugins (like waypoints for example to add classes dynamically to some div to starts some animation when they are in viewport).

It works fine for the first loaded page, but then, the scripts starts running, and the hydratation occurs at the same time, so it override the code by the new hydrated code, and the changes made by jQuery are lost.

So I would like to know if there is a way in a React component (or simple javascript) to detect when the hydratation is complete so I can then restart jQuery scripts at this time.

OF course I know it's not the best to use jQuery with React and remix, but as it takes time to recode the project, we would like the component to work during the full migration.

Thank you very much for your help :)

**Comment by machour** [comment, marked as answer]
Check [useHydrated](https://github.com/sergiodxa/remix-utils#usehydrated) from `remix-utils`
